qiskit/quantum_info/operators/symplectic/clifford_circuits.py

```diff
@@ -198,11 +201,13 @@ def _append_s(clifford, qubit):
     Returns:
         Clifford: the updated Clifford.
     """
-    x = clifford.table.X[:, qubit]
-    z = clifford.table.Z[:, qubit]
+    x = clifford.paulis.x[:, qubit]
+    z = clifford.paulis.z[:, qubit]
 
-    clifford.table.phase ^= x & z
+    count_y = clifford.paulis._count_y()
+    clifford.paulis.phase += (x & z).astype(int) * 2
     z ^= x
+    clifford.paulis.phase += clifford.paulis._count_y() - count_y
```

Comment:
Again, this feels like we could be more efficient about `_count_y` (though I get that `_count_y` will change between statements this time).

(This comment applies to most changes in this file - having `_count_y` several times means a _lot_ of iteration, and it feels rather inefficient.)